src/lib/y2network/sysconfig/interface_file.rb

```diff
@@ -0,0 +1,193 @@
+# Copyright (c) [2019] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "pathname"
+require "ipaddr"
+
+module Y2Network
+  module Sysconfig
+    # This class represents a sysconfig file containing an interface configuration
```

Comment:
An InterfaceFile does not know its file path, unlike a RoutesFile. Maybe it should be made consistent?

Reply:
Yes, you are right. Actually, I am doing a heavy refactoring on InterfaceFile to act more like the RoutesFile, including `#load` and `#path` methods as part of [adding support for multiple IPs configurations](https://trello.com/c/sNisjEg2/1117-2-support-multiple-ips-configuration).